### ISSUE TITLE
27821239: Tests run fails becuse test host is installing or uninstalling

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#15611]
#### Description

Summary:

At PSPDFKit we have a consolidated tests scheme that runs several test bundles in sequence (see image 1). We noticed that in some cases this scheme might fail between execution of the individual test bundles. There is no console log message printed and nothing that really stands on in the test log, but Xcode does show an error sheet (see image 2). 

The displayed error is as follows:

> Te request to open “com.pspdfkit.tests” failed.
> 
> The request was denied by service delegate (SBMainWorkspace) for reason: Busy ("Application "com.pspdfkit.tests" is installing or uninstalling, and cannot be launched").

com.pspdfkit.tests is the test host for one of our test bundles (probably the next one to be executed, see image 3).

Steps to Reproduce:

We were unable to reproduce this in an isolated sample. Even in our test suite this happens rarely. 

Expected Results:

Tests would run one after another without issues.

Actual Results:

The consolidated test run can fail between individual test bundle executions. 

Regression:

So far we’ve seen this fail in Xcode only once. However, on CI where we use xcodebuild to run the test suite, we have been seeing similar sporadic failures for a while. Just like in this case, the console does not have a lot of useful information. The error is just shown in Xcode UI, which leads us to believe that the previous issues might be related. 

Notes:

In addition to the aforementioned images, I’m also attaching the console log and test referenced log.
## 

Product Version: Xcode 5b5
Created: 2016-08-12T10:05:38.235530
Originated: 2016-08-12T12:05:00
Open Radar Link: http://www.openradar.me/27821239

**Comment by ArielSD** [comment]
Experiencing this exact same issue!

**Comment by dagio** [comment]
Bug also reported. Duplicate of 27386351
Open since october 😞

**Comment by dallasjohnson** [comment]
Suffering this also. Any work arounds?

**Comment by dagio** [comment]
Xcode 8.3 beta 2 solved most of the errors.

You might still have some like that :

**There was an error launching the application: The request to open "demo.XcodeUITesting" failed. (The request was denied by service delegate (SBMainWorkspace) for reason: Unspecified.)**
